pkg/stream/producer_daemon.go

```diff
@@ -94,13 +94,6 @@ type producerDaemon struct {
 	stopped    int32
 }
 
-func ResetProducerDaemons() {
```

Comment:
If we are removing these old locks, should we maybe switch the complete architecture to provide producer daemons from the context instead of using these global locks?

Reply:
Switched to appctx